web_accounts_list_checker.py

```diff
@@ -247,16 +190,136 @@ def finaloutput():
+                                             " or will run a check of the JSON file for bad detection strings.")
+parser.add_argument('-u', '--username', help='[OPTIONAL] If this param is passed then this script will perform the '
+                                             'lookups against the given user name instead of running checks against '
+                                             'the JSON file.')
+parser.add_argument('-se', '--stringerror',
+                    help="Creates a site by site file for files that do not match strings. Filenames will be 'se-("
+                         "sitename).(username)",
+                    action="store_true", default=False)
+parser.add_argument('-s', '--site', nargs='*',
+                    help='[OPTIONAL] If this parameter is passed the script will check only the named site or list of '
+                         'sites.')
+parser.add_argument('-d', '--debug', help="Enable debug output", action="store_true")
+
+args = parser.parse_args()
+
+if args.debug:
+    print('Debug output enabled')
+
+# Create the final results dictionary
+overall_results = {}
+
+# Class for colors
+if check_os() == "posix":
+    class bcolors:
+        CYAN = '\033[96m'
+        GREEN = '\033[92m'
+        YELLOW = '\033[93m'
+        RED = '\033[91m'
+        ENDC = '\033[0m'
+
+        def disable(self):
+            self.CYAN = ''
+            self.GREEN = ''
+            self.YELLOW = ''
+            self.RED = ''
+            self.ENDC = ''
+
+# if we are windows or something like that then define colors as nothing
+else:
+    class bcolors:
+        CYAN = ''
+        GREEN = ''
+        YELLOW = ''
+        RED = ''
+        ENDC = ''
+
+        def disable(self):
+            self.CYAN = ''
+            self.GREEN = ''
+            self.YELLOW = ''
+            self.RED = ''
+            self.ENDC = ''
+
+###################
+# Main
+###################
+
+# Add this in case user presses CTRL-C
+signal.signal(signal.SIGINT, signal_handler)
+
+# Suppress HTTPS warnings
+urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+# Read in the JSON file
+with open('web_accounts_list.json') as data_file:
+    data = json.load(data_file)
+
+# Check if user input for specific sites
+if args.site:
+    # cut the list of sites down to only the requested ones
+    args.site = [x.lower() for x in args.site]
+    data['sites'] = [x for x in data['sites'] if x['name'].lower() in args.site]
+    if len(data['sites']) == 0:
+        print(' -  Sorry, the requested site or sites were not found in the list')
+        sys.exit()
+    sites_not_found = len(args.site) - len(data['sites'])
+    if sites_not_found:
+        print(' -  WARNING: %d requested sites were not found in the list' % sites_not_found)
+    print(' -  Checking %d sites' % len(data['sites']))
+
+else:
+    print(' -  %s sites found in file.' % len(data['sites']))
+
+# create result file
+result_file = open("result.txt", "w")
```

Comment:
It seems to me there's no need to introduce a (temporary) output file for just collecting the outcomes from different threads.

Reply:
I saw this TODO 
```
2 - Make it so the script will append comment to the entry and output to file
```
that why I added this feature.

@WebBreacher what do you think? Should I add this or not ?

* I will create a new pull if you agree to add it .